sources/packages/web/src/components/common/Root.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <div
+    class="v-main body-background"
+    @mouseover="setLastActivityTime"
+    @click="setLastActivityTime"
+    @keyup="setLastActivityTime"
+  >
+    <slot></slot>
+    <ConfirmExtendTime
+      ref="extendTimeModal"
+      :startTimer="startTimer"
+      :clientIdType="clientIdType"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { onMounted, ref, onUnmounted, computed } from "vue";
+import { ClientIdType } from "@/types";
+import {
+  useAuth,
+  ModalDialog,
+  useInstitutionAuth,
+  useFormatters,
+} from "@/composables";
+import {
+  MINIMUM_IDLE_TIME_FOR_WARNING_SUPPORTING_USER,
+  MINIMUM_IDLE_TIME_FOR_WARNING_STUDENT,
+  MINIMUM_IDLE_TIME_FOR_WARNING_INSTITUTION,
+  MINIMUM_IDLE_TIME_FOR_WARNING_AEST,
+} from "@/constants/system-constants";
+import ConfirmExtendTime from "@/components/common/modals/ConfirmExtendTime.vue";
+
+export default {
+  components: { ConfirmExtendTime },
+  props: {
+    clientIdType: {
+      type: String,
+      required: true,
+      default: "" as ClientIdType,
+    },
+  },
+  setup(props: any) {
+    const { executeRenewTokenIfExpired } = useAuth();
+    const lastActivityLogin = ref(new Date());
+    const interval = ref();
+    const extendTimeModal = ref({} as ModalDialog<boolean>);
+    const startTimer = ref(false);
+    const { isAuthenticated } = useInstitutionAuth();
+    const { getDatesDiff } = useFormatters();
+
+    const minimumIdleTime = computed(() => {
+      switch (props.clientIdType) {
+        case ClientIdType.Student:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_STUDENT;
+        case ClientIdType.Institution:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_INSTITUTION;
+        case ClientIdType.SupportingUsers:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_SUPPORTING_USER;
+        case ClientIdType.AEST:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_AEST;
+        default:
+          return 0.02;
+      }
+    });
+
+    const startIdleCheckerTimer = () => {
+      if (isAuthenticated) {
+        /* eslint-disable */
+        interval.value = setInterval(checkIdle, 1000);
```

Comment:
It throws error when we remove eslint-disable because checkIdle has been defined after startIdleCheckerTimer.
function as const gets defined sequentially unlike function definitions.

Now the issue is, if we define checkIdle before startIdleCheckerTimer, the previous error will get resolved, but checkIdle has -> confirmExtendTimeModal

![image](https://user-images.githubusercontent.com/54600590/139509288-3afcfb31-7821-4feb-9c4e-061c6b85b508.png)

Hence if i move confirmExtendTimeModal ahead of checkIdle like
![image](https://user-images.githubusercontent.com/54600590/139509429-58ce6bf0-53a0-41d7-86f0-16534036a778.png)

Now I need to put startIdleCheckerTimer ahead CheckIdle and looks like a circular dependency. 
We can sync up to talk about it if needed.

Reply:
yes, it's because of circular dependency.  sure, we can syncup

Reply:
Currently we couldn't find a solution for this. so we can move ahead with  /* eslint-disable */.
But trying to find solution will be more ideal.